lib/slice.go

```diff
@@ -15,3 +15,42 @@ func StringSliceEqual(a, b []string) bool {
 	}
 	return true
 }
+
+// StringSliceMergeSorted takes two string slices that are assumed to be sorted
+// and does a zipper merge of the two sorted slices, removing any cross-slice
+// duplicates. If any individual slice contained duplicates those will be
+// retained.
+func StringSliceMergeSorted(a, b []string) []string {
```

Comment:
The `lib` package seems to violate the recommended practice for package names: [blog.golang.org bad package names](https://blog.golang.org/package-names#TOC_5.). 

How likely do you think it is that this function gets re-used? Maybe it can be unexported next to the one caller ?

Alternatively maybe it could go into a  `lib/stringsort` package.

Reply:
While I agree with you in general, I'd like to avoid doing a refactor like that at the same time as implementing a feature. There are more functions that just this one that would end up in a package like `lib/stringslice`.

Reply:
https://github.com/hashicorp/consul/pull/7934 (to be merged after this one)